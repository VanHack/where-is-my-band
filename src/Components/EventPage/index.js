import React from "react";
import styled from "styled-components";
import { withScriptjs, withGoogleMap, GoogleMap, Marker } from "react-google-maps"

const MyMapComponent = withScriptjs(withGoogleMap((props) =>
  <GoogleMap
    defaultZoom={12}
    defaultCenter={{ lat: props.lat, lng: props.lng }}>
    {props.isMarkerShown && <Marker position={{ lat: props.lat, lng: props.lng }} />}
  </GoogleMap>
))

const Link = styled.a`
  color: white;
`;

const Card = styled.div`
  list-style-type: none;
  background: #7b1fa2;
  color: white;
  padding: 15px;
  border-radius: 0px 3px 3px 0px;
  width: 100%;
  margin: 15px;
  transition: all 0.2s ease-in-out;

  @media (min-width: 1200px) {
    flex-basis: 40%;
    flex-grow: 1;
  }
`;
const Venue = styled.h1`
  font-size: 30px;
  color: white;
  font-weight: bold;
  margin: 0 0 15px;
`;
const City = styled.div`
  font-size: 20px;
`;
const EventDate = styled.div`
  display: inline-block;
  margin-left: auto;
  font-size: 20px;
`;
const TicketDate = styled.div`
  font-size: 20px;
`;

const EventPage = ({
  data: { venue, datetime: date, on_sale_datetime: onSale },
  artist
}) => {
  return (
    <Card>
      <Link href="/"> Back </Link> 
      <h1>{ artist.name }</h1>
      <Venue>{venue.name}</Venue>
      <City>{`${venue.city}, ${venue.country}`}</City>
      <EventDate>Date: {new Date(date).toLocaleDateString()}</EventDate>
      <TicketDate>
        Tickets available:{" "}
        {onSale ? new Date(onSale).toLocaleDateString() : " TBA"}
      </TicketDate>
      <MyMapComponent
        isMarkerShown
        lat={parseFloat(venue.latitude)}
        lng={parseFloat(venue.longitude)}
        googleMapURL="https://maps.googleapis.com/maps/api/js?v=3.exp"
        loadingElement={<div style={{ height: `100%` }} />}
        containerElement={<div style={{ height: `200px` }} />}
        mapElement={<div style={{ height: `100%` }} />}
      />
    </Card>
  );
};

export default EventPage;
