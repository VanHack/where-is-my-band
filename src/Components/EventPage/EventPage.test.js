import React from 'react';
import EventPage from './index';
import renderer from 'react-test-renderer';

test('renders the component according to snapshot', () => {
  const myEvent = {
    venue: {
      name: 'MyVenue',
      datetime: new Date(),
      on_sale_datetime: new Date()
    }
  }
  
  const rockStar = {
    image_url: 'http://example.com/some-image.jpg',
    name: 'Rock Star!',
    excerpt: 'Rock Star band description',
    facebook_page_url: 'http://facebook.com/rockstar',
  }

  const component = renderer.create(
    <EventPage data={myEvent} artist={rockStar}/>,
  );

  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
