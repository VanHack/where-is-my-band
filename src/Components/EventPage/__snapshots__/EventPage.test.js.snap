// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the component according to snapshot 1`] = `
<div
  className="sc-bwzfXH jfwgnS"
>
  <a
    className="sc-bdVaJa gvVfeb"
    href="/"
  >
     Back 
  </a>
  <h1>
    Rock Star!
  </h1>
  <h1
    className="sc-htpNat cSTQBi"
  >
    MyVenue
  </h1>
  <div
    className="sc-bxivhb drhNVP"
  >
    undefined, undefined
  </div>
  <div
    className="sc-ifAKCX bWmygj"
  >
    Date: 
    Invalid Date
  </div>
  <div
    className="sc-EHOje jEDgFc"
  >
    Tickets available:
     
     TBA
  </div>
  <br />
  <div
    style={
      Object {
        "height": "100%",
      }
    }
  />
</div>
`;
